spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -122,7 +149,47 @@ def refresh_from_pdb(self, pdb_state):
         if 'var_properties' in pdb_state:
             self.set_var_properties(pdb_state['var_properties'])
 
+    def set_pdb_state(self, pdb_state):
+        """Set current pdb state."""
+        if pdb_state is not None and isinstance(pdb_state, dict):
+            self.refresh_from_pdb(pdb_state)
+
+    def pdb_continue(self):
+        """Continue debugging."""
+        # Run Pdb continue to get to the first breakpoint
+        # Fixes 2034
+        self.pdb_execute('continue')
+
     # ---- Private API (overrode by us) ----------------------------
+    def _readline_callback(self, line):
+        """Callback used when the user inputs text in stdin."""
+        if not self.is_waiting_pdb_input():
+            if self._input_ready:
+                # This is a regular input call
+                self._finalize_input_request()
```

Comment:
This is maybe something that should be changed in qtconsole as well. I will open an issue there.

Reply:
https://github.com/jupyter/qtconsole/issues/370